ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java

```diff
@@ -1289,6 +1338,9 @@ GetOption getGetOption() {
      */
     @Override
     MessageFormatFlags getOperationFlag() {
+      if (mdCacheLookupResult == mdCacheLookUpStatus.mdCacheHit) {
+        return MessageFormatFlags.BlobInfo;
```

Comment:
It looks like if we the blob ID in metadata cache, we are modifying GetBlobOperation to get blobInfo instead of blob for the first chunk to validate the blob. 

Just thinking that one other way could be to fetch the getBlobInfo from [NonBlockingRouter](https://github.com/linkedin/ambry/blob/master/ambry-router/src/main/java/com/github/ambry/router/NonBlockingRouter.java#L240) (a bit higher level) itself if the blob ID is found in cache. If the getBlobInfo() is successful, we could issue getBlob() call. Else, we could report the error to user. With this approach, advantage seems to be that we don't need to integrate BlobInfo operation inside GetBlobOperation and handle the exceptions at upper level. But, we might have to do the operations we do during handling of metadata response such as calculating the needed [chunk IDs](https://github.com/linkedin/ambry/blob/master/ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java#L1523), [initializing the data chunk objects](https://github.com/linkedin/ambry/blob/master/ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java#L1571), etc in the [GetBlobOperation constructor](https://github.com/linkedin/ambry/blob/master/ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java#L154) since there is no request-response flow for the metadata chunk. What do you think?